test/junit/scala/tools/nsc/reporters/ConsoleReporterTest.scala

```diff
@@ -0,0 +1,188 @@
+      reporter.display(noPos, "Testing display", reporter.INFO)
+      assertEquals("Testing display\n", err_out.toString)
+      err_out.reset
+
+      reporter.display(noPos, "Testing display", reporter.WARNING)
+      assertEquals("warning: Testing display\n", err_out.toString)
+      err_out.reset
+
+      reporter.display(noPos, "Testing display", reporter.ERROR)
+      assertEquals("error: Testing display\n", err_out.toString)
+      err_out.reset
+
+      // display test with DefinedPosition
+      reporter.display(posWithSource, "Testing display", reporter.INFO)
+      assertEquals(source + ":1: Testing display\n" + content + "\n    ^\n", err_out.toString)
+      err_out.reset
+
+      reporter.display(posWithSource, "Testing display", reporter.WARNING)
+      assertEquals(source + ":1: warning: Testing display\n" + content + "\n    ^\n", err_out.toString)
+      err_out.reset
+
+      reporter.display(posWithSource, "Testing display", reporter.ERROR)
+      assertEquals(source + ":1: error: Testing display\n" + content + "\n    ^\n", err_out.toString)
+      err_out.reset
+
+      reporter.ERROR.count = 50
+      reporter.display(posWithSource, "Testing display for maxerrs to pass", reporter.ERROR)
+      assertEquals(source + ":1: error: Testing display for maxerrs to pass\n" + content + "\n    ^\n", err_out.toString)
+      err_out.reset
+
+      reporter.ERROR.count = 101
+      reporter.display(posWithSource, "Testing display for maxerrs to fail", reporter.ERROR)
+      assertEquals("", err_out.toString)
+      err_out.reset
+
+      reporter.WARNING.count = 25
+      reporter.display(posWithSource, "Testing display for maxwarns to pass", reporter.WARNING)
+      assertEquals(source + ":1: warning: Testing display for maxwarns to pass\n" + content + "\n    ^\n", err_out.toString)
+      err_out.reset
+
+      reporter.WARNING.count = 50
+      reporter.display(posWithSource, "Testing display for maxwarns to fail", reporter.WARNING)
+      assertEquals("", err_out.toString)
+      err_out.reset
+    }
+
+
+    @Test
+    def finishTest(): Unit = {
+      reporter.ERROR.count = 0
+      reporter.WARNING.count = 0
+      reporter.finish
+      assertEquals("", err_out.toString)
+      err_out.reset
+
+      reporter.WARNING.count = 3
+      reporter.ERROR.count = 10
+      reporter.finish
+      assertEquals("three warnings found\n10 errors found\n", err_out.toString)
+      err_out.reset
+    }
+
+    @Test
+    def displayPromptTest(): Unit = {
+      val stack = new ByteArrayOutputStream
+      System.setErr(new PrintStream(stack))  // used for checking the stack trace of displayPrompt()
+      
+      reporter.displayPrompt;
+      assertFalse(stack.toString.equals(""))
+      stack.reset
+
+      {
+        val input = new StringReader("r\n")
+        val input_reader = new BufferedReader(input)
+        val reporter = new ConsoleReporter(new Settings(), input_reader, output_writer)
+        reporter.displayPrompt
```

Comment:
Not a huge deal, but I'd expect parens to demonstrate a side-effecting application. That's the standard advice. This looks like it just accesses a value.